### PR TITLE
refactor: use mustache template in manifest

### PR DESCRIPTION
**Pull request by nliu-ms**
1) mustache template in manifest
2) Move remote teams app id to app studio plugin config. Rename remoteTeamsAppId -> teamsAppId
image

Note:
1) local teams app id has been moved to localSettings.json
2) Only azure tab has been supported

### CODE REVIEW

**Review by qinezh — APPROVED** [review]

**Review by dooriya — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/appstudio/constants.ts [hunk 1]
```diff
@@ -237,6 +237,24 @@ export const STATIC_TABS_TPL: IStaticTab[] = [
   },
 ];
 
+export const STATIC_TABS_TPL_FOR_MULTI_ENV: IStaticTab[] = [
+  {
+    entityId: "index",
+    name: "Personal Tab",
+    contentUrl: "{{{profile.fx-resource-frontend-hosting.endpoint}}}/index.html#/tab",
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/appstudio/constants.ts:244]: triple "{" & "}" ?

File: packages/fx-core/src/plugins/resource/appstudio/constants.ts [hunk 2]
```diff
@@ -119,8 +119,8 @@ export const TEAMS_APP_MANIFEST_TEMPLATE_FOR_MULTI_ENV = `{
   ],
   "validDomains": [],
   "webApplicationInfo": {
-      "id": "{appClientId}",
-      "resource": "{webApplicationInfoResource}"
+      "id": "{{profile.fx-resource-aad-app-for-teams.clientId}}",
+      "resource": "{{{profile.fx-resource-aad-app-for-teams.applicationIdUris}}}"
```
Comment (dooriya) [inline packages/fx-core/src/plugins/resource/appstudio/constants.ts:123]: I see you're using triple `'{'` and `'}'` in some placeholders, while mustache uses double `'{'` and `'}'`for the tag to be resolved.

**Comment by nliu-ms** [comment]
@dooriya triple `{}`s for `/` in URLs will be escaped by default.

Reference: 
All variables are HTML-escaped by default. If you want to render unescaped HTML, use the triple mustache: {{{name}}}. You can also use & to unescape a variable.

**Review by jayzhang — APPROVED** [review]

**Comment by dooriya** [comment]
> @dooriya triple `{}`s for `/` in URLs will be escaped by default.
> 
> Reference:
> All variables are HTML-escaped by default. If you want to render unescaped HTML, use the triple mustache: {{{name}}}. You can also use & to unescape a variable.

Good to know that. Thanks for the explanation.

**Pull request merged**